sway-ir/src/instruction.rs

```diff
@@ -71,58 +63,48 @@ pub enum Instruction {
         asset_id: Value,
         gas: Value,
     },
-    /// Reading a specific element from an array.
-    ExtractElement {
-        array: Value,
-        ty: Type,
-        index_val: Value,
-    },
-    /// Reading a specific field from (nested) structs.
-    ExtractValue {
-        aggregate: Value,
-        ty: Type,
-        indices: Vec<u64>,
-    },
     /// Umbrella instruction variant for FuelVM-specific instructions
     FuelVm(FuelVmInstruction),
     /// Return a local variable.
     GetLocal(LocalVar),
-    /// Writing a specific value to an array.
-    InsertElement {
-        array: Value,
-        ty: Type,
-        value: Value,
-        index_val: Value,
-    },
-    /// Writing a specific value to a (nested) struct field.
-    InsertValue {
-        aggregate: Value,
-        ty: Type,
-        value: Value,
-        indices: Vec<u64>,
+    /// Translate a pointer from a base to a nested element in an aggregate type.
+    GetElemPtr {
+        base: Value,
+        elem_ptr_ty: Type,
```

Comment:
No, this will always be a pointer type.  If you create one with `Instruction::get_elem_ptr()` you pass in the base elem type, but it will be made into a pointer and stored as one.  This is because `Instruction::get_type()` on the `GetElemPtr` variant needs to return a pointer.